src/log.cpp

```diff
@@ -124,6 +109,37 @@ std::ostream actionstream(&action_buf);
 std::ostream infostream(&info_buf);
 std::ostream verbosestream(&verbose_buf);
 
+// Android
+#ifdef __ANDROID__
+
+static unsigned int g_level_to_android[] = {
+	ANDROID_LOG_INFO,     // LL_NONE
+	//ANDROID_LOG_FATAL,
+	ANDROID_LOG_ERROR,    // LL_ERROR
+	ANDROID_LOG_WARN,     // LL_WARNING
+	ANDROID_LOG_WARN,     // LL_ACTION
+	//ANDROID_LOG_INFO,
+	ANDROID_LOG_DEBUG,    // LL_INFO
+	ANDROID_LOG_VERBOSE,  // LL_VERBOSE
+};
+
+class AndroidSystemLogOutput : public ICombinedLogOutput {
+	public:
+		AndroidSystemLogOutput()
+		{
+			g_logger.addOutput(this);
+		}
+		void log_raw(LogLevel lev, const std::string &line)
+		{
+			assert(ARRLEN(g_level_to_android) == LL_MAX);
```

Comment:
Hmm, okay then, I guess this could be done at a later time.

I just realized minetest doesn't have a static assert macro - to make one you'd have the condition known at compile-time expand into an expression that's either 1 or -1 as the size of an array being declared: i.e. `int somearray[ARRLEN(g_level_to_android) == LL_MAX ? 1 : -1]`